### PR TITLE
sanitized text in input/IP & port fields on input change callback

### PR DESCRIPTION
**Pull request by fernando-cortez**
Jira task [here](https://unity3d.atlassian.net/browse/GOMPS-274?atlOrigin=eyJpIjoiZTgyMmUwMzU2ODgzNDY3OWI5MGNmNGU4YTQyZDFhMjMiLCJwIjoiaiJ9).

This does a sanitize pass on our input fields for IP/room name & port number. We only allow alphanumerics and/or a set of allowed characters (in the IP field, we allow '.'). The sanitize check is hooked up to the UI element's OnValueChanged callback field on the PopupPanel prefab.

This does not necessarily sanitize the fields for the proper IP format, but instead removes invalid characters, added whitespaces, etc.

Edit: this should allow for copy-pasting room codes!

### CODE REVIEW

**Review by Cosmin-B — DISMISSED** [review]

File: Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs [hunk 1]
```diff
@@ -59,6 +59,8 @@ public class PopupPanel : MonoBehaviour
 
         private const string k_DefaultConfirmText = "OK";
 
+        static readonly char[] k_InputFieldIncludeChars = new[] { '.' };
```
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs:62]: Did you test this with relay? I think you're missing '_' no?

Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs]: Done.

**Review by Cosmin-B — APPROVED** [review]

File: Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs [hunk 2]
```diff
@@ -122,6 +124,45 @@ private void OnConfirmClick()
...
+        /// <param name="dirtyString"> string to sanitize. </param>
+        /// <param name="includeChars"> Array of chars to include. </param>
+        /// <returns> Sanitized text string. </returns>
+        static string Sanitize(string dirtyString, char[] includeChars = null)
+        {
+            var result = new StringBuilder(dirtyString.Length);
+            foreach (char c in dirtyString)
+            {
+                if (char.IsLetterOrDigit(c) ||
+                    (includeChars != null && Array.Exists(includeChars, excludeChar => excludeChar == c)))
```
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs:139]: exclude --> include

Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/UI/PopupPanel.cs]: Done.

**Review by Cosmin-B — DISMISSED** [review]

**Comment by Cosmin-B** [comment]
@fernando-cortez please resolve the conflicts before merging

**Review by Cosmin-B — APPROVED** [review]

**Review by SamuelBellomo — APPROVED** [review]

**Pull request merged**